control/grpc.py

```diff
@@ -122,6 +124,24 @@ def get_hosts_with_dhchap_key(self, subsys):
             return self.host_dhchap_key[subsys]
         return {}
 
+    def add_host_nqn(self, subsys, hostnqn):
+        if not subsys in self.host_nqn:
+            self.host_nqn[subsys] = []
+        self.host_nqn[subsys].append(hostnqn)
+
+    def remove_host_nqn(self, subsys, hostnqn):
+        if not subsys in self.host_nqn:
+            return
+        try:
+            self.host_nqn[subsys].remove(hostnqn)
+        except ValueError:
+            pass
+
+    def get_host_count(self, subsys):
+        if not subsys in self.host_nqn:
+            return 0
+        return len(self.host_nqn[subsys])
```

Comment:
```suggestion
        return len(self.host_nqn[subsys])
```

This is safe for defaultdict

Reply:
I still keep the first "if", otherwise an empty set is added for "subsys".